modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java

```diff
@@ -561,7 +574,8 @@ private boolean initme() {
             }
         }
         userName = (String) parameters.get(USERNAME);
-        password = (String) parameters.get(PASSWORD);
+        password = resolveSecureVaultExpressions((String) parameters.get(PASSWORD));
```

Comment:
What if password parameter is not encrypted? Can resolveSecureVaultExpressions method handle that too?